drivers/hmis/app/models/hmis/form/instance_project_match.rb

```diff
@@ -0,0 +1,62 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+class Hmis::Form::InstanceProjectMatch
+  include Memery
+  attr_accessor :project, :instance
+
+  # match types ordered by rank. Lower rank is better
+  RANKED_MATCHES = [
+    PROJECT_MATCH = 'project'.freeze,
+    ORGANIZATION_MATCH = 'organization'.freeze,
+    PROJECT_TYPE_AND_FUNDER_MATCH = 'project_type_and_funder'.freeze,
+    PROJECT_TYPE_MATCH = 'project_type'.freeze,
+    PROJECT_FUNDER_MATCH = 'project_funder'.freeze,
+    DEFAULT_MATCH = 'default'.freeze,
+  ].freeze
+  MATCH_RANKS = RANKED_MATCHES.each_with_index.to_h.freeze
+
+  def initialize(instance:, project:)
+    self.instance = instance
+    self.project = project
+  end
+
+  def rank
+    MATCH_RANKS[match]
+  end
+
+  def valid?
+    !!match
+  end
+
+  protected
+
+  # match to project. Order is significant, should return the best ranked match
```

Comment:
this could use another set of eyes. I've tried to capture the logic previously in `instance. detect_best_instance_scope_for_project`

Reply:
thanks for setting this up and parsing through all that logic! I think it is much more readable. Left a few comments but otherwise looks good.

Reply:
I haven't tried running it yet on your branch but https://github.com/greenriver/hmis-warehouse/blob/release-101/drivers/hmis/spec/models/hmis/form/definition_spec.rb should have decent coverage.

Reply:
Thanks! Yes that spec passes for me